lib/Event/Model/AssetPreLoadEvent.php

```diff
@@ -0,0 +1,58 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Event\Model;
+
+use Pimcore\Event\Traits\ArgumentsAwareTrait;
+use Pimcore\Model\Asset;
+use Pimcore\Model\Exception\NotFoundException;
+use Symfony\Contracts\EventDispatcher\Event;
+
+class AssetPreLoadEvent extends Event implements ElementEventInterface
+{
+    use ArgumentsAwareTrait;
+
+    protected ?Asset $asset;
+
+    /**
+     * AssetEvent constructor.
+     *
+     * @param array $arguments additional parameters (e.g. "versionNote" for the version note)
+     */
+    public function __construct(?Asset $asset, array $arguments = [])
+    {
+        $this->asset = $asset;
+        $this->arguments = $arguments;
+    }
+
+    public function getAsset(): ?Asset
+    {
+        if (empty($this->document)) {
+            throw new NotFoundException();
+        }
+        return $this->asset;
+    }
```

Comment:
This cannot return `null`
```suggestion
    public function getAsset(): Asset
    {
        if (empty($this->document)) {
            throw new NotFoundException();
        }
        return $this->asset;
    }
```
Same for the other events.

Reply:
same here, it has to be nullable otherwise I don't see any point to hook in to return "null" in global way which ends in a return "null" from the GetByID()

Reply:
But when it's null, an exception will be thrown. So `null` will never be returned.

Reply:
yeah thats true, so for this explicit method. I will adapt it 👍